automerge-frontend/src/state_tree/resolved_path.rs

```diff
@@ -48,152 +52,101 @@ impl std::fmt::Debug for Target {
-            Target::List(listtarget) => listtarget.multivalue.default_value(&self.root.objects),
-            Target::Text(texttarget) => texttarget.multivalue.default_value(&self.root.objects),
-            Target::Counter(countertarget) => {
-                countertarget.multivalue.default_value(&self.root.objects)
+            Target::Table(tabletarget) => tabletarget.multivalue.default_value(),
+            Target::List(listtarget) => listtarget.multivalue.default_value(),
+            Target::Text(texttarget) => texttarget.multivalue.default_value(),
+            Target::Counter(countertarget) => countertarget.multivalue.default_value(),
+            Target::Primitive(p) => p.multivalue.default_value(),
+            Target::Character(ctarget) => {
+                Value::Primitive(Primitive::Str(ctarget.multivalue.default_grapheme()))
             }
-            Target::Primitive(p) => p.multivalue.default_value(&self.root.objects),
-            Target::Character(ctarget) => ctarget.multivalue.default_value(&self.root.objects),
         }
     }
 
     pub fn values(&self) -> std::collections::HashMap<amp::OpId, Value> {
         match &self.target {
-            Target::Map(maptarget) => maptarget.multivalue.realise_values(&self.root.objects),
+            Target::Map(maptarget) => maptarget.multivalue.realise_values(),
             Target::Root(root) => {
                 let mut result = std::collections::HashMap::new();
                 result.insert(random_op_id(), root.root.value());
                 result
             }
-            Target::Table(tabletarget) => tabletarget.multivalue.realise_values(&self.root.objects),
-            Target::List(listtarget) => listtarget.multivalue.realise_values(&self.root.objects),
-            Target::Text(texttarget) => texttarget.multivalue.realise_values(&self.root.objects),
-            Target::Counter(countertarget) => {
-                countertarget.multivalue.realise_values(&self.root.objects)
-            }
-            Target::Primitive(p) => p.multivalue.realise_values(&self.root.objects),
-            Target::Character(ctarget) => ctarget.multivalue.realise_values(&self.root.objects),
+            Target::Table(tabletarget) => tabletarget.multivalue.realise_values(),
+            Target::List(listtarget) => listtarget.multivalue.realise_values(),
+            Target::Text(texttarget) => texttarget.multivalue.realise_values(),
+            Target::Counter(countertarget) => countertarget.multivalue.realise_values(),
+            Target::Primitive(p) => p.multivalue.realise_values(),
+            Target::Character(ctarget) => ctarget.multivalue.realise_values(),
         }
     }
 
     pub fn object_id(&self) -> Option<amp::ObjectId> {
         match &self.target {
-            Target::Map(maptarget) => Some(maptarget.value.object_id.clone()),
+            Target::Map(maptarget) => Some(maptarget.multivalue.default_object_id().unwrap()),
```

Comment:
Can we add a field to `Target` for object ID to avoid these unwraps?

Reply:
Yes, done